powertools-parameters/powertools-parameters-ssm/src/main/java/software/amazon/lambda/powertools/parameters/ssm/SSMProvider.java

```diff
@@ -124,33 +107,6 @@ public String getValue(String key) {
         return client.getParameter(request).parameter().value();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public SSMProvider defaultMaxAge(int maxAge, ChronoUnit unit) {
-        super.defaultMaxAge(maxAge, unit);
-        return this;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public SSMProvider withMaxAge(int maxAge, ChronoUnit unit) {
```

Comment:
I pulled this up to the base class. The only two transformations you can do here are common to all - `withMaxAge` and `withTransformation` - so there's no need to override and reimplement this everywhere.

I would like to change this so it returns some other object - e.g. `GetQueryBuilder` or so - so we have a natural scope for the mutation of the parameters associated with the get, and don't have to mutate and un-mutate the Baseprovider itself (see `BaseProvider::resetToDefaults::`)

I see this as a subsequent change almost worthy of an RFC and not something to do here - this is already a massive PR. 

```
    protected void resetToDefaults() {
        cacheManager.resetExpirationTime();
        if (transformationManager != null) {
            transformationManager.setTransformer(null);
        }
    }
```